substrate/frame/support/src/lib.rs

```diff
@@ -1064,1232 +961,941 @@ pub mod pallet_prelude {
+	/// 	}
+	/// }
+	///
+	/// #[import_section(events)]
+	/// #[frame_support::pallet]
+	/// mod pallet {
+	/// # 	use frame_support::pallet_prelude::*;
+	/// #
+	/// 	#[pallet::pallet]
+	/// 	pub struct Pallet<T>(_);
+	/// #
+	/// # 	#[pallet::config]
+	/// # 	pub trait Config: frame_system::Config {
+	/// # 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+	/// # 	}
+	/// }
+	/// ```
+	///
+	/// This will result in the contents of `some_section` being _verbatim_ imported into
+	/// the pallet above. Note that since the tokens for `some_section` are essentially
+	/// copy-pasted into the target pallet, you cannot refer to imports that don't also
+	/// exist in the target pallet, but this is easily resolved by including all relevant
+	/// `use` statements within your pallet section, so they are imported as well, or by
+	/// otherwise ensuring that you have the same imports on the target pallet.
+	///
+	/// It is perfectly permissible to import multiple pallet sections into the same pallet,
+	/// which can be done by having multiple `#[import_section(something)]` attributes
+	/// attached to the pallet.
+	///
+	/// Note that sections are imported by their module name/ident, and should be referred to
+	/// by their _full path_ from the perspective of the target pallet.
+	pub use frame_support_procedural::import_section;
+
+	/// Allows defining getter functions on `Pallet` storage.
+	///
+	/// ## Example
+	///
+	/// ```
+	/// #[frame_support::pallet]
+	/// mod pallet {
+	/// # 	use frame_support::pallet_prelude::*;
+	/// #
+	/// 	#[pallet::pallet]
+	/// 	pub struct Pallet<T>(_);
+	///
+	/// 	#[pallet::storage]
+	/// 	#[pallet::getter(fn my_getter_fn_name)]
+	/// 	pub type MyStorage<T> = StorageValue<_, u32>;
+	/// #
+	/// # 	#[pallet::config]
+	/// # 	pub trait Config: frame_system::Config {}
+	/// }
+	/// ```
+	///
+	/// See [`pallet::storage`](`frame_support::pallet_macros::storage`) for more info.
+	pub use frame_support_procedural::getter;
+
+	/// Allows generating the `Store` trait for all storages.
+	///
+	/// DEPRECATED: Will be removed, do not use.
+	/// See <https://github.com/paritytech/substrate/pull/13535> for more details.
+	///
+	/// To generate a `Store` trait associating all storages, annotate your `Pallet` struct
+	/// with the attribute `#[pallet::generate_store($vis trait Store)]`, e.g.:
+	///
+	/// ```ignore
+	/// #[pallet::pallet]
+	/// #[pallet::generate_store(pub(super) trait Store)]
+	/// pub struct Pallet<T>(_);
+	/// ```
+	/// More precisely, the `Store` trait contains an associated type for each storage. It is
+	/// implemented for `Pallet` allowing access to the storage from pallet struct.
+	///
+	/// Thus when defining a storage named `Foo`, it can later be accessed from `Pallet` using
+	/// `<Pallet as Store>::Foo`.
+	pub use frame_support_procedural::generate_store;
+
+	/// Allows you to define some extra constants to be added into constant metadata.
```

Comment:
nit: Defines constants that are added to the constant metadata field [link to frame-metadata] https://github.com/paritytech/frame-metadata/blob/94e7743fa454963609763cf9cccbb7f85bc96d2f/frame-metadata/src/v15.rs#L267 or a similar page describing the metadata concept/how to fetch it?